### PR TITLE
Feature/compression

### PR DESCRIPTION
**Pull request by northwestwitch**
Moved and tidied compression modules and created their tests.

### CODE REVIEW

File: lib/MIP/Program/Compression/Gzip.pm [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    my $FILEHANDLE;
+    my $stderrfile_path;
+    my $stderrfile_path_append;
+    my $stdout;
+    my $decompress;
+    my $infile_path;
+    my $outfile_path;
+
+    my $tmpl = {
+        FILEHANDLE      => { store       => \$FILEHANDLE },
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Compression/Gzip.pm]: It is good to have some order in the template and the comments to the sub. Could you use this order: hash_ref, array_ref, scalar_ref, scalar (required option), scalar (optional), scalar(with defaults). WE shoudl record this in the documentation under a subroutines section

File: lib/MIP/Program/Compression/Pigz.pm [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            allow       => [ 0, 1 ],
+            strict_type => 1,
+            store       => \$verbose
+        },
+    };
+
+    check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
+
+    ## pigz
+    my @commands = qw{pigz};    #Stores commands depending on input parameters
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Compression/Pigz.pm]: Move comment to line above

Comment (henrikstranneheim) [reply lib/MIP/Program/Compression/Pigz.pm]: and #Stores -> # Stores

File: lib/MIP/Program/Compression/Pigz.pm [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+        push @commands, q{--verbose};
+    }
+
+    if ($decompress) {
+
+        push @commands, q{--decompress};
+    }
+
+    if ($stdout) {    #Write to stdout stream
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Compression/Pigz.pm]: Move comment to line above

File: lib/MIP/Program/Compression/Pigz.pm [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+        push @commands, q{--stdout};
+    }
+
+    ## FILE
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Compression/Pigz.pm]: FILE -> # Infile

File: lib/MIP/Program/Compression/Gzip.pm [hunk 5]
```diff
@@ -60,15 +60,15 @@ sub gzip {
...
-    my $stderrfile_path_append;
     my $stdout;
     my $decompress;
     my $outfile_path;
+    my $FILEHANDLE;
+    my $stderrfile_path;
+    my $stderrfile_path_append;
 
     my $tmpl = {
-        quiet => {
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Compression/Gzip.pm]: Can you move the default variables to the end within the template and header. Usually they are the one you care the least about

File: lib/MIP/Program/Compression/Pigz.pm [hunk 6]
```diff
@@ -41,17 +41,17 @@ sub pigz {
...
-## Arguments: $FILEHANDLE, $stdout, $stderrfile_path, $stderrfile_path_append, $decompress, $infile_path, $outfile_path, $processes, $quiet, $verbose
-##          : $FILEHANDLE             => Filehandle to write to
-##          : $stderrfile_path        => Stderrfile path
-##          : $stderrfile_path_append => Append stderr info to file path
+## Arguments:  $quiet, $verbose, $infile_path, $stdout, $decompress,  $outfile_path, $processes, $FILEHANDLE, $stderrfile_path, $stderrfile_path_append
+##          : $quiet                  => Suppress all warnings
+##          : $verbose                => Verbosity
+##          : $infile_path            => Infile path
 ##          : $stdout                 => Write on standard output, keep original files unchanged
 ##          : $decompress             => Decompress
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Compression/Pigz.pm]: Move default variables to the end of the template and the sub header

File: lib/MIP/Program/Compression/Gzip.pm (hunk 5 above)
Comment (northwestwitch) [reply lib/MIP/Program/Compression/Gzip.pm]: I moved $FILEHANDLE, $stderrfile_path, $stderrfile_path_append at the end in Gzip and Pigz. 
stdout in this case is not really a default one so I've put it after the required one (infile_path). 
In the templated (Command_module.pm) I just have another variable before these standard ones, stdoutfile_path, which is also a standard one. I've put them in the order:
stdoutfile_path and then the 3 above.

**Pull request merged**